### DISCUSSION TITLE
How to create a ORM representation of a "query" ?

### DISCUSSION THREAD

**Discussion by ikus060** [sqlalchemy/sqlalchemy#7626] [Usage Questions] [answered]
Hello,

I'm having trouble to create an Object to represent the result of a complex query.  The general idea is as follow. I have `DhcpRecord` defining `ip` and DnsRecord also defining `ip`. I want to create a union between the two type of record to get a list of all IP address.

Here the model:
```
class DhcpRecord(Base):
    ip = Column(String, nullable=False, unique=True)
    mac = Column(String, nullable=False, unique=True)

class DnsRecord(Base):
    name = Column(String, unique=False, nullable=False)
    type = Column(String, nullable=False)
    ttl = Column(Integer, nullable=False, default=3600)
    value = Column(String, nullable=False)
```

To get the result I want, I build the following query:
```
select(DhcpRecord.ip).union(
            select(DnsRecord.value).filter(DnsRecord.type == 'A'))
```

But the results are only tuple. How to I get a list of object. And how could I make it even more dynamic to retrieve related records.
e.g.:

```
class IpView(???):
  ip = Column(String)
  related_dhcp_records = relationship(...)
  related_dns_records = relationship(...)
```

Basically, I'm looking to create an object representation of the UNION query.

**Comment by ikus060** [comment]
I tried with something similar to the follower but it obviously not working.
```
        q1 = select(DhcpRecord.ip.label('ip')).union(
            select(DnsRecord.value.label('ip')).filter(DnsRecord.type == 'A'))
        q2 = self.session.execute(
            select(aliased(DhcpRecord, adapt_on_names=True)).from_statement(q1))
        result = self.session.execute(q2)
```

**Comment by CaselIT** [comment]
I believe this is what you are looking for
https://docs.sqlalchemy.org/en/14/orm/nonstandard_mappings.html#mapping-a-class-against-arbitrary-subqueries

**Reply by zzzeek** [reply]
mmm i think the user should use aliased() here

**Comment by zzzeek** [comment, marked as answer]
so the "map a class to a select" is a thing, yes, but you're trying to use aliased().  to get ORM objects from a union using new style statements is documented at https://docs.sqlalchemy.org/en/14/tutorial/data_select.html#selecting-orm-entities-from-unions

**HOWEVER**, what you are actually doing there, getting DhcpRecord and DnsRecord in the *same* list of objects would not be possible unless you change your mapping.  You would want to use [mapping inheritance](https://docs.sqlalchemy.org/en/14/orm/inheritance.html) here so that you could load these two types of objects what we call "polymorphically".  if you want to keep the identical table structure you'd want to use [concrete table inheritance](https://docs.sqlalchemy.org/en/14/orm/inheritance.html#concrete-table-inheritance) and you then could apply the above "union" form to the supertype of DhcpRecord and DnsRecord.
